### DISCUSSION TITLE
Should I manually update LSPs that have been auto-installed by nvim-lspconfig?

### DISCUSSION THREAD

**Discussion by heapifyman** [LazyVim/LazyVim#2513] [Q&A] [answered]
I have added Angular LSP as described [in the documentation](https://www.lazyvim.org/configuration/examples) - see below. angular-ls was automatically installed by mason.

If I now run `:Mason` in neovim, it tells me there is an update of angularls

> ◍ angular-language-server angularls   new version available: 17.0.3 -> 17.1.1

Should I update this manually from the mason UI or will lazyvim eventually notice the update an install it automatically?

---

`~/.config/nvim/lua/plugins/lsp-config.lua`:

```
return {
  "neovim/nvim-lspconfig",
  opts = {
    servers = {
      angularls = {},
      ...
    }
  }
}
```

**Comment by dpetka2001** [comment, marked as answer]
Just press `U` in the Mason UI to update everything that has an update. LazyVim doesn't have access to what Mason is doing. Mason refreshes the mason-registry and checks for updates every time you open the Mason UI.

**Reply by heapifyman** [reply]
Ok. Thanks for the quick reply. I thought maybe that would be done automatically when pressing `S` in the LazyVim UI.

Do you know of a way to automate mason updates? For LazyVim someone suggested `nvim --headless '+Lazy! sync' +qa`. Is there something similar for mason updates?

**Reply by dpetka2001** [reply]
No idea. I just periodically open the Mason UI and see whatever is available for updating. You should direct this question in Mason repo Discussions section, as the maintainer will have more insight if something like this is possible.
